source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+        return { slope, intercept };
+    };
+
+    cv::Mat pointCloudToCvBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+    std::vector<cv::Point2f> estimate2DMarkerCenters(const aruconano::Marker &markers)
+    {
+        if(markers.empty())
+        {
+            std::cout << "No markers detected";
+            return {};
+        }
+
+        std::vector<cv::Point2f> markerCenters2D;
+        markerCenters2D.reserve(markers.size());
+
+        const auto corner0 = markers[0];
+        const auto corner1 = markers[1];
+        const auto corner2 = markers[2];
+        const auto corner3 = markers[3];
+
+        // Fit line between two diagonal corners
+        const auto backDiagonal{ fitLine(corner2, corner0) };
+        const auto forwardDiagonal{ fitLine(corner3, corner1) };
+
+        // Find intersection of the two lines
+        // a1*x + b1 = a2*x + b2
+        // x = (b2-b1) / (a1-a2)
+        const auto xCoord =
+            (forwardDiagonal.intercept - backDiagonal.intercept) / (backDiagonal.slope - forwardDiagonal.slope);
+        const auto yCoord = backDiagonal.slope * xCoord + backDiagonal.intercept;
+        markerCenters2D.emplace_back(cv::Point2f(xCoord, yCoord));
+
+        return markerCenters2D;
+    }
+
+    std::vector<cv::Point2f> get2DMarkerCorners(const aruconano::Marker &markers)
+    {
+        if(markers.empty())
+        {
+            std::cout << "No markers detected";
+            return {};
+        }
+
+        std::vector<cv::Point2f> markerCorners2D;
+        markerCorners2D.reserve(markers.size() * 4);
+
+        markerCorners2D.emplace_back(markers[0]);
+        markerCorners2D.emplace_back(markers[1]);
+        markerCorners2D.emplace_back(markers[2]);
+        markerCorners2D.emplace_back(markers[3]);
+
+        return markerCorners2D;
+    }
+
+    std::vector<cv::Point3d> estimate3DMarkerPoints(const Zivid::PointCloud &pointCloud,
+                                                    const std::vector<cv::Point2f> &markerPoints2D)
+    {
+        if(markerPoints2D.empty())
+        {
+            return {};
+        }
+
+        const int width = pointCloud.width();
+        const int height = pointCloud.height();
+        const auto points = pointCloud.copyPointsXYZ();
+
+        std::vector<cv::Point3d> markerPoints3D;
+        markerPoints3D.reserve(markerPoints2D.size());
+        for(const auto &point2D : markerPoints2D)
+        {
+            const float x = point2D.x;
+            const float y = point2D.y;
```

Comment:
Not a must, but perhaps a good idea. I would consider using 'u' and 'v' for image (2D) coordinates, and 'x' 'y' 'z' for 3D coordinates.